File: code/1686. Stone Game VI.cpp
class Solution {
public:
    int stoneGameVI(vector<int>& aliceValues, vector<int>& bobValues) {
        vector<vector<int>> v;
        int n = aliceValues.size();
        for (int i = 0; i < n; i++) {
            v.push_back({aliceValues[i] + bobValues[i], aliceValues[i], bobValues[i]});
        }
        sort(begin(v), end(v), [&](vector<int>& a, vector<int>& b){
            return a[0] > b[0];
        });
        int alice = 0, bob = 0;
        for (int i = 0; i < n; i++) {
            if (i % 2 == 0) {
                alice += v[i][1];
            } else {
                bob += v[i][2];
            }
        }
        return alice >= bob ? (alice == bob ? 0 : 1) : -1;
    }
};

File: code/1686. Stone Game VI.py
class Solution:
    def stoneGameVI(self, aliceValues: List[int], bobValues: List[int]) -> int:
        v = []
        for i in range(len(aliceValues)):
            v.append([-(aliceValues[i] + bobValues[i]), -aliceValues[i], -bobValues[i]])
        v.sort(key=lambda x : x[0])
        alice = 0
        bob = 0
        for i, (tot, a, b) in enumerate(v):
            #print(i, -tot, -a, -b)
            if i % 2 == 0:
                alice += -a
            else:
                bob += -b
        if alice == bob:
            return 0
        elif alice > bob:
            return 1
        else:
            return -1
        